kate/src/com.rs

```diff
@@ -383,11 +384,9 @@ pub fn build_commitments(
 	let (prover_key, _) = public_params.trim(block_dims.cols).map_err(Error::from)?;
 	let row_eval_domain = EvaluationDomain::new(block_dims.cols).map_err(Error::from)?;
 
-	let mut result_bytes: Vec<u8> = Vec::new();
-	result_bytes.reserve_exact(PROVER_KEY_SIZE * extended_rows_num);
-	unsafe {
-		result_bytes.set_len(PROVER_KEY_SIZE * extended_rows_num);
-	}
+	// @internal: `PROVER_KEY_SIZE` must match with the output size of `Commitment::to_bytes`.
+	const_assert_eq!(G1Affine::SIZE, PROVER_KEY_SIZE);
```

Comment:
In which circumstances can this assertion fail? If the config is wrong, the assertion will fail but should we handle that case more gracefully instead of shutting down the node?

Reply:
This assertion is verified at **compilation time**, so if both constants are different the _node will not compile_. See [static_assertions crate](https://docs.rs/static_assertions/1.1.0/static_assertions/index.html)
It has 0 overhead  at runtime.

Reply:
Thanks! Didn't know that.